File: NiukeAB/AB7/AB7.py

class Quene(object):
    def __init__(self):
        self.s1=list()
        self.s2=list()
    def push(self, x):
        self.s1.append(x)
    def pop(self):
        if  len(self.s2)>0:
            return self.s2.pop()
        if len(self.s1)==0:
            return "error"
        for i in range(len(self.s1)):
            self.s2.append(self.s1.pop())
        return self.s2.pop()
    def front(self):
        if len(self.s2)>0:
            return self.s2[-1]
        elif len(self.s1)>0:
            return self.s1[0]
        else:
            return "error"
    def check(self, x):
        if x=='pop':
            print(self.pop())
        elif x=='front':
            print(self.front())
        else:
            self.push(int(x.split()[-1]))
        


n=input()
tmp=Quene()
for _ in range(int(n)):
    tmp.check(input())

File: NiukeAB/AB7/AB7.go
package main

import "fmt"
type Quene struct{
	 s1 []string
	 s2 []string
}
func (q *Quene) Push(x string) {
	q.s1=append(q.s1,x)
}
func (q *Quene) Pop() (x string) {
	if len(q.s2)>0{
		x=q.s2[len(q.s2)-1]
		q.s2=q.s2[:len(q.s2)-1]
		return 
	}
	if len(q.s1)<1{
		return "error"
	}
	x=q.s1[0]
	q.s1=q.s1[1:]
	for i:=len(q.s1)-1;i>-1;i--{
		q.s2=append(q.s2,q.s1[i])
	}
	q.s1=q.s1[:0]
	return 
}
func (q *Quene) Front() (x string) {
	if len(q.s2)>0{
		return q.s2[len(q.s2)-1]
	}
	if len(q.s1)<1{
		return "error"
	}
	return q.s1[0]
}
func main() {
	var n int
	fmt.Scan(&n)
	var q Quene
	for i := 0; i < n; i++ {
		var op, x string
		fmt.Scan(&op)
		if op == "push" {
			fmt.Scan(&x)
			q.Push(x)
		} else if op == "pop" {
			x = q.Pop()
			fmt.Println(x)
		} else if op == "front" {
			x = q.Front()
			fmt.Println(x)
		}
	}		
}